### ISSUE TITLE
timezone-less NWB files are failing DANDI-side validation

### ISSUE THREAD

**Issue by bendichter** [dandi/dandi-schema#284]
We have a server-side issue with validation of NWB files. **Until this is resolved, most NWB files submitted to DANDI will be unpublishable.**

NWB recently removed the requirement for NWB session_start_time to have a timezone ([PR](https://github.com/NeurodataWithoutBorders/pynwb/pull/1886)). We previously had been defaulting to assigning the timezone of where the conversion was run, but were concerned that this was often incorrect. It’s also very unusual for recording systems or any other metadata to include the timezone, so the timezone requirement made it challenging to automate conversion.

The necessary changes have been made to the software, and these new timezone-less files propagate just fine through NWB GUIDE, NeuroConv, PyNWB, HDMF, NWB Inspector, and the dandi CLI. Creation, validation, and upload work well.

However, these new files are failing validation on DANDI server side, which means it is currently impossible to publish these files. See [here](https://gui-staging.dandiarchive.org/dandiset/213840) for an example.

Our preferred solution would be to relax the validation on DANDI server side to allow for NWB files that have datetime that do not have timezones. We're not sure where this line would be. The schema just lists the requirement as `datetime`, and it's not clear how that is meant to be serealized/deserialized and if that requires a timezone to be present.

https://github.com/dandi/dandi-schema/blob/e135307e38e8fccf83676cf1d6faba9b142a8de0/dandischema/models.py#L1186

**Comment by rly** [comment]
Correct, we have reverted NWB back to requiring timezones if a time is present. We will be allowing date-only datetimes, but I believe that should pass validation fine here.

**Comment by satra** [comment]
perhaps we can do what this group did here: https://github.com/equinor/fmu-dataio/pull/1184/files#diff-cae7076f47e60fab21815e69b08afb7b606aae4893bb9096fc01c6905e0543b5

**Comment by satra** [comment]
@candleindark - please see the example i pointed to above. that may be a good solution to consider.

**Comment by satra** [comment]
> Are you proposing the following:
>1. We mark the datetime fields in Pydantic models to the format of "iso-date-time" in the generated JSON schema
>2. Chain an ISO datetime validator to the python-jsonschema validator as a format checker to check for the custom "iso-date-time" format

yes, if we don't consolidate validation into one form (see #331 - let's continue the validation discussion there)
